frontend/__tests__/context/PlaygroundReducer.test.tsx

```diff
@@ -0,0 +1,540 @@
+        type: "ADD_INPUT_DOCUMENT" as const,
+        payload: newDocument,
+      };
+      const newState = playgroundReducer(initialState, action);
+
+      expect(newState.inputDocuments).toHaveLength(1);
+      expect(newState.inputDocuments[0]).toEqual(newDocument);
+    });
+  });
+
+  describe("UPDATE_INPUT_DOCUMENT", () => {
+    it("should update an existing input document", () => {
+      const existingDocument = {
+        id: "test-id",
+        name: "Original Name",
+        content: "Original content",
+      };
+      const initialState = createTestState({
+        inputDocuments: [existingDocument],
+      });
+
+      const updatedDocument = {
+        id: "test-id",
+        name: "Updated Name",
+        content: "Updated content",
+      };
+      const action = {
+        type: "UPDATE_INPUT_DOCUMENT" as const,
+        payload: updatedDocument,
+      };
+      const newState = playgroundReducer(initialState, action);
+
+      expect(newState.inputDocuments).toHaveLength(1);
+      expect(newState.inputDocuments[0]).toEqual(updatedDocument);
+    });
+
+    it("should not modify other documents when updating one", () => {
+      const document1 = {
+        id: "id-1",
+        name: "Document 1",
+        content: "Content 1",
+      };
+      const document2 = {
+        id: "id-2",
+        name: "Document 2",
+        content: "Content 2",
+      };
+      const initialState = createTestState({
+        inputDocuments: [document1, document2],
+      });
+
+      const updatedDocument = {
+        id: "id-1",
+        name: "Updated Document 1",
+        content: "Updated Content 1",
+      };
+      const action = {
+        type: "UPDATE_INPUT_DOCUMENT" as const,
+        payload: updatedDocument,
+      };
+      const newState = playgroundReducer(initialState, action);
+
+      expect(newState.inputDocuments).toHaveLength(2);
+      expect(newState.inputDocuments[0]).toEqual(updatedDocument);
+      expect(newState.inputDocuments[1]).toEqual(document2);
+    });
+  });
+
+  describe("REMOVE_INPUT_DOCUMENT", () => {
+    it("should remove an input document by id", () => {
+      const document1 = {
+        id: "id-1",
+        name: "Document 1",
+        content: "Content 1",
+      };
+      const document2 = {
+        id: "id-2",
+        name: "Document 2",
+        content: "Content 2",
+      };
+      const initialState = createTestState({
+        inputDocuments: [document1, document2],
+      });
+
+      const action = {
+        type: "REMOVE_INPUT_DOCUMENT" as const,
+        payload: "id-1",
+      };
+      const newState = playgroundReducer(initialState, action);
+
+      expect(newState.inputDocuments).toHaveLength(1);
+      expect(newState.inputDocuments[0]).toEqual(document2);
+    });
+
+    it("should clear output documents when removing an input document", () => {
+      const document = {
+        id: "id-1",
+        name: "Document 1",
+        content: "Content 1",
+      };
```

Comment:
Seems like a lot of copy and paste for this document (+ other types), can we refactor up this file to make it more readable for what the tests are focused on?